pkgs/applications/editors/vscode/generic.nix

```diff
@@ -69,6 +69,16 @@ in
     dontBuild = true;
     dontConfigure = true;
 
+    patchPhase =
+      if system != "x86_64-darwin" then ''
```

Comment:
Can you elaborate why this is not required on darwin?

Reply:
Does the darwin flavor of sudo-shell even use the same scripts to achieve administrator access? How does nix even handle sandboxing shell commands on an OS like that?

Reply:
As I suspected:

https://github.com/jorangreef/sudo-prompt/blob/master/index.js

sudo-prompt uses a completely different code path on macOS/Darwin to achieve the sudo prompt, going as far as bundling a special program in a zip file, unpacking it, and running it, to elevate the requested command.

The above patch for `/bin/bash` and `/usr/bin/pkexec` processes only applies to NixOS, and any other Linux system adhering to Nix file paths for these two utilities.

I wasn't sure if this was the correct way to limit it, though. I took the hint from the install process, which gates Darwin specific install process from everything else, since only Darwin will be installing a single .app bundle.

Reply:
I would prefer if we apply this unconditionally to prevent bit rot as long as it does not break something not already broken on darwin.

Reply:
Sure, you can apply it unconditionally, but upstream only supplies binary packages for x86_64 anyway.